src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -1282,16 +1282,16 @@ public bool IsYinYangGloballyBound(int n)
             }
 
 #if DEBUG
-            public void AssertValidYinYangBounds(int n, ref VBuffer<float> features, VBuffer<float>[] centroids)
+            public void AssertValidYinYangBounds(int n, in VBuffer<float> features, VBuffer<float>[] centroids)
```

Comment:
>VBuffer<float>[] centroids [](start = 84, length = 26)

also "in"?

Reply:
It's an array of VBuffers. We pass the array like any other reference.

---
In reply to: [229506248](https://github.com/dotnet/machinelearning/pull/1454#discussion_r229506248) [](ancestors = 229506248)